app/bundles/CoreBundle/Update/Step/PreUpdateChecksStep.php

```diff
@@ -38,6 +38,14 @@ public function shouldExecuteInFinalStage(): bool
      */
     public function execute(ProgressBar $progressBar, InputInterface $input, OutputInterface $output): void
     {
+        /**
+         * We can't easily fetch data about an update package without unzipping it first, so for now
+         * we skip the pre-update checks if the user manually provides an update package.
+         */
+        if (!empty($input->getOption('update-package'))) {
+            return;
+        }
```

Comment:
A bug I came across while working on this PR - if the user provides an update package themselves, things break because the pre-update checks don't expect this